core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -593,6 +596,14 @@ QservRestrictorPlugin::_newSecIndexRestrictor(
     query::QsRestrictor::Ptr p;
     std::shared_ptr<query::ColumnRef> secIndex =
         resolveAsColumnRef(context, cp->left);
+    if (secIndex)  {
+        std::string theDirTable =
+            context.cssFacade->getDirTable(secIndex->db, secIndex->table);
+        if ( ! theDirTable.empty() ) {
+            LOGF_DEBUG("Setting dirTable for sIndex to %1%" % theDirTable);
+            secIndex->table = theDirTable;
+        }
+    }
```

Comment:
Tell me more about secondary index db, I think it is the first time I hear we have/support it!
 * I checked https://dev.lsstcorp.org/trac/wiki/db/Qserv/CSS and it is not mentioned
 * I grepped for it in qserv_testdata/case0?/data/*.cfg and we are not setting it there
 * I grepped for dirD it in qserv source code and didn't get any match either (dirC and dirT are found)

Agreed wrt not checking if column name matches. I'll fix that.

Agreed wrt IN

Regarding resolveAsColumnRef - yup, it does call copyAsColumnRef. I did run some tests with valgrind and saw no memory issues whatsoever.